File: components/toggle-dark-mode/toggle-dark-mode.clj
(defn main [file files file-list-idx _]
  {:depends-on [{:type :js :src  "/components/toggle-dark-mode/toggle-dark-mode.js"}
                {:type :css :src "/components/toggle-dark-mode/toggle-dark-mode.css"}]
   :body [:div.toggle-dark-mode-container
          [:button.toggle-dark-mode ""]]})

File: jake-site-deploy/components/ToggleDarkMode/ToggleDarkMode.ts
const ToggleDarkMode = () => ({
  dependsOn: [
    {
      src: "/components/ToggleDarkMode/toggle-dark-mode.js",
    },
    {
      src: "/components/ToggleDarkMode/toggle-dark-mode.css",
    }
  ],
  body: ["div", { class: 'toggle-dark-mode-container' },
         ["button", { class: 'toggle-dark-mode' }, ""]],
})

export default ToggleDarkMode;
